### PR TITLE
Pirate asteroid map

### PR DESCRIPTION
**Pull request by GutsySS13**
This map can be merged, it still needs some random loot to be added and it's final use as yet is undetermined. 

I recommend it to be a good event map or with loot for an exploration map. You guys can decide. but it seems a bit redundant as a separate fork so may as well merge and then people can change the map as they desire and we can remove the fork.

We may also want to remove all the separate asteroid field maps as I don't think they are going to be used any more due to other map issues and the potential new warp (or whatever) system that's coming.

### CODE REVIEW

**Comment by lmas** [comment]
:+1: Fire it away.
Could you explain more about the map issues and the new warp system stuff?

**Comment by GutsySS13** [comment]
Well kwask was talking about having 1 map where people just flew across instead of multiple space maps made into tiles to save the case where now if someone gets fired across space they keep alive a lot of empty space maps in memory. that way there's only one space zone and when they've crossed enough to reach their destination they arrive at it. It would save a lot of instances and issues we were having with maps.

(My idea at the time was to have a sort of hyperspace zone where people enter warp space and travel around and can then warp back at preset places to say get to the asteroid or other maps, i'm not sure how eva suits would get to places though.)

This map could be used as an exploration map or as a admin event map. It's a pirate base. 

-- Meta alert --

This pirate asteroid map info:
Basically some pirates have setup a base on a an asteroid and have been looting passing ships for supplies, unknown to them a bunch of hivebots have also been inhabiting the asteroid also scavenging passing ships. The current state is that there has been some sort of altercation between the bots and the pirates and there's been a big fight and the pirates lost. The hive bots have been building a high tech spaceship to leave the asteroid but it's not finished yet. (players aren't supposed to know this).

**Comment by Kwask** [comment]
Don't bother with removing maps, they're only a few dozen kilobytes at most, and you're shooting yourself in the foot if you end up needing them later. As far as the warp system is concerned, there will still be random locations, and we can still have a large variety of them. The way it would work is the game would randomly select optional maps to load on round-start, so it won't load all of them at all times.

**Comment by lmas** [comment]
That sounds good, looking forward to finally be able to explore random sectors filled with content.

 I think we should put such warp maps in it's own subdir to differentiate them from the usual maps that's alway loaded.

**Comment by Kwask** [comment]
@lmas, even though you just disabled one of the sectors that has content in it yesterday? :p

**Comment by lmas** [comment]
Well yeah duh! It's never used, it get's the boot. Seriously though, was there even a way to travel to that place ICly?

**Comment by Kwask** [comment]
You could travel to it the same way as you can travel to any other unknown sector. Every unknown sector has a 10% chance to appear as an "unidentified object" on the paper that lists all known sectors.

I've seen a few people stumble across that sector and then even one antag found the hidden hand teleporter and used it to get back to the station

**Comment by lmas** [comment]
~~Well shit. Looks like someone will have to revert again.~~ Fixed with d0e239df62d794536e1ce4c9b90b83ffac0bf547.

**Comment by GutsySS13** [comment]
the problem with the astfield*[1-3].dmm files are that they are broken and probably need to be re-made. I found a bug in later maps with the asteroid templates I was using that some of the asteroids were turning into sandstone (not quite sure what apollo patch did this) and haven't been back to the older maps to check so it's probably better to just remove them. They were not that hard to generate anyway as would now probably take just an hour or two to make anyway if you really wanted to me to make random blobs of rock floating in space and were just there to liven up the floating through empty space experience.

**Comment by lmas** [comment]
It's up to you then on what to do with them. 

Personally I find that big empty space to be really boring, even if it's supposed to be realistic or something, and could need an upgrade to make it more fun.

Like you said, a hyperwarp zone where you could easily travel to a new sector (where each map tile on that map equals one sector you can warp back to) sounds like a better idea. Add in a scanner of some kind so you can detect interesting things in the sectors (and don't have to zip around, warp back and explore each and every sector manually to find something) so you don't miss the fun stuff.
